staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -136,16 +136,23 @@ func TestGetServerGroupsWithTimeout(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		// first we need to write headers, otherwise http client will complain about
 		// exceeding timeout awaiting headers, only after we can block the call
-		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Connection", "keep-alive")
+		if wf, ok := w.(http.Flusher); ok {
+			wf.Flush()
+		}
 		<-done
+		w.WriteHeader(http.StatusOK)
```

Comment:
is this actually needed if we flush? Doesn't flush end the header phase and start the body?

Reply:
According to [docs](https://golang.org/pkg/net/http/#Flusher) it flushes data, so no matter what was in there, it sends all.

Reply:
Discussed with Stefan on IRC, I'm gonna drop this line, since we don't care what will happen after we've cancelled.